internal/status/gateway.go

```diff
@@ -40,24 +40,35 @@ func UpdateGatewayStatusProgrammedCondition(gw *gwapiv1b1.Gateway, svc *corev1.S
 			}
 		}
 
-		var gwAddrs []gwapiv1b1.GatewayAddress
+		if svc.Spec.ClusterIP != "" {
+			addresses = append(addresses, svc.Spec.ClusterIP)
+		}
+		if len(svc.Spec.ClusterIPs) != 0 {
```

Comment:
redundant check?

Reply:
One is clusterip, another is clusterips.

Reply:
done